pkgs/applications/editors/vim/plugins/update.py

```diff
@@ -109,10 +118,86 @@ def plugin2nix(self, pdesc: PluginDesc, plugin: pluginupdate.Plugin) -> str:
   }};
 
 """.format(
-        buildFn="buildNeovimPluginFrom2Nix" if isNeovim else "buildVimPluginFrom2Nix", plugin=plugin, src_nix=src_nix, repo=repo)
+            buildFn="buildNeovimPluginFrom2Nix"
+            if isNeovim
+            else "buildVimPluginFrom2Nix",
+            plugin=plugin,
+            src_nix=src_nix,
+            repo=repo,
+        )
         print(content)
         return content
 
+
+def get_rust_dep_plugins() -> List[str]:
+    """Get list of plugins that have Rust dependencies."""
+    overrides = (Path(__file__).parent / "overrides.nix").read_text().splitlines()
+    stack: Deque[str] = deque()
+    first_plugins_line = next(
+        line_num
+        for line_num, line in enumerate(overrides, 1)
+        if line == "self: super: {"
+    )
+    last_plugins_line = 0
+
+    for line_num, line in enumerate(
+        overrides[first_plugins_line - 1 :], first_plugins_line
+    ):
+        for char in line:
+            if char == "{":
+                stack.append(char)
+            elif char == "}":
+                stack.pop()
+
+        if len(stack) == 0:
+            last_plugins_line = line_num
+            break
+
+    plugins = overrides[first_plugins_line:last_plugins_line]
+
+    plugin_def_lines = {
+        line for line in plugins if re.search(r"^\s\s[a-zA-Z0-9-_]*\s=", line)
```

Comment:
That is not quite right. There can also be two == or none.

Reply:
Basically here I am looking for lines like `clang_complete = super.clang_complete.overrideAttrs (old: {`. In what use case would it be `something ==`? And for the none case I don't know why that would apply either.

Reply:
I was thinking about a different file format. Derp.